src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -239,20 +225,48 @@ create_distributed_table(PG_FUNCTION_ARGS)
 
 	relation_close(relation, NoLock);
 
-	char *distributionColumnName = text_to_cstring(distributionColumnText);
-	Assert(distributionColumnName != NULL);
+	bool shardCountIsStrict = false;
+	if (distributionColumnText)
+	{
+		if (PG_ARGISNULL(2))
+		{
+			PG_RETURN_VOID();
+		}
 
-	char distributionMethod = LookupDistributionMethod(distributionMethodOid);
+		int shardCount = ShardCount;
+		if (!PG_ARGISNULL(4))
+		{
+			if (pg_strncasecmp(colocateWithTableName, "default", NAMEDATALEN) != 0 &&
```

Comment:
could be nice to use IsColocateWithNone and IsColocateWithDefault

Reply:
moved these to https://github.com/citusdata/citus/pull/6745 and addressed there